test/e2e/self_hosted.go

```diff
@@ -161,6 +163,15 @@ func SelfHostedSpec(ctx context.Context, inputGetter func() SelfHostedSpecInput)
 			return selfHostedClusterProxy.GetClient().Get(ctx, client.ObjectKey{Name: "kube-system"}, kubeSystem)
 		}, "5s", "100ms").Should(BeNil(), "Failed to assert self-hosted API server stability")
 
+		preMoveMachineList := &clusterv1alpha3.MachineList{}
```

Comment:
Can we add a comment like in clusterctl_upgrade?

Let's please use the v1beta struct here. We don't have to use v1alpha3 here so it would be nice to avoid it

(I know this means we have to have our own version of machinesMatch, but I would prefer that over using v1alpha3 here)